### PR TITLE
Fix typos detected by github.com/client9/misspell

### PR DESCRIPTION
**Pull request by seratch**
Fixing typos is sometimes very hard. It's not so easy to visually review them. Recently, I discovered a very useful tool for it, [misspell](https://github.com/client9/misspell). 

This pull request fixes minor typos detected by [misspell](https://github.com/client9/misspell) except for the false positives. If you would like me to work on other files as well, let me know. 

## before 

```
$ misspell . | grep -v 'LocalStrings'
java/javax/servlet/resources/XMLSchema.dtd:36:32: "overriden" is a misspelling of "overridden"
java/javax/servlet/resources/XMLSchema.dtd:173:43: "overriden" is a misspelling of "overridden"
java/javax/servlet/resources/j2ee_web_services_client_1_1.xsd:191:34: "refered" is a misspelling of "referred"
java/javax/servlet/resources/javaee_web_services_client_1_2.xsd:263:34: "refered" is a misspelling of "referred"
java/javax/servlet/resources/javaee_web_services_client_1_3.xsd:173:34: "refered" is a misspelling of "referred"
java/javax/servlet/resources/javaee_web_services_client_1_4.xsd:169:34: "refered" is a misspelling of "referred"
java/javax/servlet/resources/web-app_3_0.xsd:129:61: "occurence" is a misspelling of "occurrence"
java/javax/servlet/resources/web-app_4_0.xsd:125:61: "occurence" is a misspelling of "occurrence"
java/javax/servlet/resources/web-app_3_1.xsd:129:61: "occurence" is a misspelling of "occurrence"
java/javax/servlet/resources/web-fragment_3_0.xsd:129:61: "occurence" is a misspelling of "occurrence"
java/javax/servlet/resources/web-fragment_3_1.xsd:129:61: "occurence" is a misspelling of "occurrence"
java/javax/servlet/resources/web-jsptaglibrary_1_1.dtd:38:58: "prefered" is a misspelling of "preferred"
java/javax/servlet/resources/web-fragment_4_0.xsd:125:61: "occurence" is a misspelling of "occurrence"
java/javax/servlet/resources/web-jsptaglibrary_1_2.dtd:53:62: "prefered" is a misspelling of "preferred"
java/javax/servlet/resources/web-jsptaglibrary_2_0.xsd:685:50: "prefered" is a misspelling of "preferred"
java/org/apache/catalina/connector/CoyoteInputStream.java:141:25: "transfered" is a misspelling of "transferred"
java/org/apache/catalina/connector/InputBuffer.java:354:25: "transfered" is a misspelling of "transferred"
java/org/apache/catalina/filters/ExpiresFilter.java:1173:7: "occurence" is a misspelling of "occurrence"
java/org/apache/catalina/ha/session/SessionMessageImpl.java:74:54: "transfered" is a misspelling of "transferred"
java/org/apache/catalina/ha/session/SessionMessageImpl.java:147:74: "TRANSFERED" is a misspelling of "TRANSFERRED"
java/org/apache/catalina/ha/session/DeltaManager.java:299:22: "transfered" is a misspelling of "transferred"
java/org/apache/catalina/ha/session/DeltaManager.java:805:42: "transfered" is a misspelling of "transferred"
java/org/apache/catalina/ha/session/DeltaManager.java:806:59: "transfered" is a misspelling of "transferred"
java/org/apache/catalina/ha/session/DeltaManager.java:814:34: "transfered" is a misspelling of "transferred"
java/org/apache/catalina/ha/session/DeltaManager.java:826:38: "transfered" is a misspelling of "transferred"
java/org/apache/catalina/ha/session/DeltaManager.java:1200:48: "transfered" is a misspelling of "transferred"
java/org/apache/catalina/ha/session/DeltaManager.java:1334:44: "transfered" is a misspelling of "transferred"
java/org/apache/catalina/ha/session/DeltaManager.java:1372:86: "TRANSFERED" is a misspelling of "TRANSFERRED"
java/org/apache/catalina/ha/session/DeltaManager.java:1373:31: "TRANSFERED" is a misspelling of "TRANSFERRED"
java/org/apache/catalina/ha/tcp/SimpleTcpCluster.java:799:32: "transfered" is a misspelling of "transferred"
java/org/apache/catalina/ha/session/mbeans-descriptors.xml:262:36: "transfered" is a misspelling of "transferred"
java/org/apache/catalina/servlets/CGIServlet.java:1164:67: "existant" is a misspelling of "existent"
java/org/apache/catalina/servlets/CGIServlet.java:1428:59: "recommedations" is a misspelling of "recommendations"
java/org/apache/catalina/storeconfig/StandardContextSF.java:164:47: "orginal" is a misspelling of "original"
java/org/apache/catalina/storeconfig/StoreAppender.java:295:18: "orginal" is a misspelling of "original"
java/org/apache/catalina/tribes/transport/bio/BioSender.java:104:39: "occured" is a misspelling of "occurred"
java/org/apache/catalina/users/mbeans-descriptors.xml:46:26: "persistant" is a misspelling of "persistent"
java/org/apache/catalina/valves/StuckThreadDetectionValve.java:356:22: "agains" is a misspelling of "against"
java/org/apache/coyote/Response.java:101:21: "commited" is a misspelling of "committed"
java/org/apache/coyote/Response.java:255:15: "commited" is a misspelling of "committed"
java/org/apache/coyote/Response.java:336:12: "commited" is a misspelling of "committed"
java/org/apache/coyote/Response.java:615:8: "commited" is a misspelling of "committed"
java/org/apache/el/Messages_es.properties:18:17: "Problemas" is a misspelling of "Problems"
java/org/apache/el/lang/VariableMapperFactory.java:26:27: "momento" is a misspelling of "memento"
java/org/apache/tomcat/dbcp/dbcp2/DriverManagerConnectionFactory.java:86:20: "propeties" is a misspelling of "properties"
java/org/apache/tomcat/dbcp/dbcp2/DriverManagerConnectionFactory.java:92:58: "propeties" is a misspelling of "properties"
java/org/apache/tomcat/dbcp/dbcp2/DriverManagerConnectionFactory.java:98:23: "propeties" is a misspelling of "properties"
java/org/apache/tomcat/dbcp/dbcp2/managed/TransactionContextListener.java:31:14: "commited" is a misspelling of "committed"
java/org/apache/tomcat/dbcp/dbcp2/managed/TransactionContextListener.java:34:72: "commited" is a misspelling of "committed"
java/org/apache/tomcat/dbcp/dbcp2/managed/ManagedConnection.java:207:93: "commited" is a misspelling of "committed"
java/org/apache/tomcat/dbcp/dbcp2/managed/TransactionContext.java:70:30: "compatability" is a misspelling of "compatibility"
java/org/apache/tomcat/dbcp/dbcp2/managed/TransactionRegistry.java:64:30: "compatability" is a misspelling of "compatibility"
java/org/apache/tomcat/util/bcel/classfile/ClassParser.java:32:27: "appropiate" is a misspelling of "appropriate"
java/org/apache/tomcat/util/bcel/classfile/ClassParser.java:144:26: "implicitely" is a misspelling of "implicitly"
java/org/apache/tomcat/util/buf/CharChunk.java:293:15: "substract" is a misspelling of "subtract"
java/org/apache/tomcat/util/buf/CharChunk.java:301:15: "substract" is a misspelling of "subtract"
java/org/apache/tomcat/util/buf/ByteChunk.java:392:15: "substract" is a misspelling of "subtract"
java/org/apache/tomcat/util/buf/ByteChunk.java:408:15: "substract" is a misspelling of "subtract"
java/org/apache/tomcat/util/buf/ByteChunk.java:426:21: "transfered" is a misspelling of "transferred"
java/org/apache/tomcat/util/buf/ByteChunk.java:433:15: "substract" is a misspelling of "subtract"
java/org/apache/tomcat/util/digester/RulesBase.java:100:29: "accidently" is a misspelling of "accidentally"
java/org/apache/tomcat/util/http/fileupload/ParameterParser.java:122:43: "presense" is a misspelling of "presence"
java/org/apache/tomcat/util/http/fileupload/MultipartStream.java:608:38: "preceeded" is a misspelling of "preceded"
java/org/apache/tomcat/util/digester/package.html:1031:0: "accessable" is a misspelling of "accessible"
java/org/apache/tomcat/util/net/SecureNioChannel.java:390:71: "supressed" is a misspelling of "suppressed"
res/rat/rat-excludes.txt:54:34: "ther" is a misspelling of "there"
res/findbugs/filter-false-positives.xml:1010:26: "behavoiur" is a misspelling of "behaviour"
test/org/apache/catalina/tribes/demos/MapDemo.java:139:56: "paramters" is a misspelling of "parameters"
test/org/apache/tomcat/util/descriptor/web/TestWebXmlOrdering.java:331:34: "existant" is a misspelling of "existent"
webapps/docs/config/context.xml:816:28: "applicaiton" is a misspelling of "application"
webapps/docs/config/http.xml:569:27: "addtional" is a misspelling of "additional"
webapps/docs/config/http.xml:582:27: "addtional" is a misspelling of "additional"
webapps/docs/changelog.xml:664:24: "specifing" is a misspelling of "specifying"
webapps/docs/host-manager-howto.xml:114:26: "retreive" is a misspelling of "retrieve"
webapps/docs/html-host-manager-howto.xml:114:28: "retreive" is a misspelling of "retrieve"
webapps/docs/monitoring.xml:1007:59: "accessable" is a misspelling of "accessible"
webapps/examples/WEB-INF/classes/jsp2/examples/simpletag/FindBookSimpleTag.java:34:56: "Tolkein" is a misspelling of "Tolkien"
```

## after

```
$ misspell . | grep -v 'LocalStrings'
java/org/apache/el/Messages_es.properties:18:17: "Problemas" is a misspelling of "Problems"
java/org/apache/el/lang/VariableMapperFactory.java:26:27: "momento" is a misspelling of "memento"
java/org/apache/tomcat/util/buf/CharChunk.java:293:15: "substract" is a misspelling of "subtract"
java/org/apache/tomcat/util/buf/CharChunk.java:301:15: "substract" is a misspelling of "subtract"
java/org/apache/tomcat/util/buf/ByteChunk.java:392:15: "substract" is a misspelling of "subtract"
java/org/apache/tomcat/util/buf/ByteChunk.java:408:15: "substract" is a misspelling of "subtract"
java/org/apache/tomcat/util/buf/ByteChunk.java:426:21: "transfered" is a misspelling of "transferred"
java/org/apache/tomcat/util/buf/ByteChunk.java:433:15: "substract" is a misspelling of "subtract"
webapps/examples/WEB-INF/classes/jsp2/examples/simpletag/FindBookSimpleTag.java:34:56: "Tolkein" is a misspelling of "Tolkien"
```

### CODE REVIEW

**Comment by markt-asf** [comment]
Thanks for this. Working through a review of the patch now.
Note that there are some files that we can't change such as the Oracle provided XSDs.

File: java/org/apache/tomcat/dbcp/dbcp2/managed/ManagedConnection.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/**
+vi /**
```
Comment (seratch) [inline java/org/apache/tomcat/dbcp/dbcp2/managed/ManagedConnection.java]: I will remove this after your review...

File: java/org/apache/catalina/ha/session/SessionMessageImpl.java [hunk 2]
```diff
@@ -144,7 +144,7 @@ public String getEventTypeString()
             case EVT_GET_ALL_SESSIONS : return "SESSION-GET-ALL";
             case EVT_SESSION_DELTA : return "SESSION-DELTA";
             case EVT_ALL_SESSION_DATA : return "ALL-SESSION-DATA";
-            case EVT_ALL_SESSION_TRANSFERCOMPLETE : return "SESSION-STATE-TRANSFERED";
+            case EVT_ALL_SESSION_TRANSFERCOMPLETE : return "SESSION-STATE-TRANSFERRED";
```
Comment (seratch) [inline java/org/apache/catalina/ha/session/SessionMessageImpl.java]: If I should keep the typo for now, I will revert this change.

File: java/org/apache/tomcat/dbcp/dbcp2/managed/ManagedConnection.java (hunk 1 above)
Comment (markt-asf) [reply java/org/apache/tomcat/dbcp/dbcp2/managed/ManagedConnection.java]: No worries. The git repo is only a mirror so I have to manually apply to the patch to svn anyway. I'd already found this one and fixed it.

File: java/org/apache/catalina/ha/session/SessionMessageImpl.java (hunk 2 above)
Comment (markt-asf) [reply java/org/apache/catalina/ha/session/SessionMessageImpl.java]: This one is fine. It is only used for logging. And the similar issue in DeltaManager should be OK too.

**Pull request closed without merging**